src/Workspaces/Core/Portable/Workspace/Host/DocumentService/Extensions.cs

```diff
@@ -1,25 +1,29 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
 namespace Microsoft.CodeAnalysis.Host
 {
     internal static class Extensions
     {
-        public static bool CanApplyChange(this TextDocument document)
+        public static bool CanApplyChange([NotNullWhen(returnValue: true)] this TextDocument? document)
```

Comment:
Are those attributes for static analysis or enforced at runtime?

Reply:
This helps the compiler nullable analysis. They are not enforced at runtime.

Reply:
And to explain what this annotation is good for: if you wrote:

```
if (CanApplyChange(textDocumentThatMightBeNull))
{
    // use textDocumentThatMightBeNull
}
```

the compiler knows that if you're inside the if block, the parameter must have been non-null so it's safe to use textDocument inside of the if.